tools/points-service/main.go

```diff
@@ -474,6 +481,37 @@ func insertOptInWithVault(db *sql.DB, logger *slog.Logger, pubkey, adder, vaultA
 	}
 }
 
+func insertManualValRecord(db *sql.DB,
+	logger *slog.Logger,
+	pubkey,
+	adder string,
+	optedInBlock uint64,
+) error {
+	rwLock.RLock()
+	defer rwLock.RUnlock()
+
+	var existingPubkey string
+	err := db.QueryRow(`
+		SELECT pubkey FROM validator_records
```

Comment:
This select is not needed, we can check err after insertion, like this:

`if strings.Contains(err.Error(), "UNIQUE constraint failed") || 
           strings.Contains(err.Error(), "duplicate key value")`

Reply:
Here I'm trying to enforce uniqueness exclusively for the `pubkey`. The current table has `UNIQUE(pubkey, adder, opted_in_block)` which seems to only enforce uniqueness for a tuple of those values.

Your suggestion is valid, but it'd require adding a new uniqueness constraint to the existing db, which seems like a riskier change than the current naive logic